src/fsharp/DetupleArgs.fs

```diff
@@ -360,14 +360,19 @@ let checkTS = function
 /// explicit tuple-structure in expr 
 let rec uncheckedExprTS expr = 
     match expr with 
-    | Expr.Op (TOp.Tuple tupInfo, _tys, args, _) when not (evalTupInfoIsStruct tupInfo) -> 
+    | Expr.Op (TOp.Tuple _, _, args, _) -> 
         TupleTS (List.map uncheckedExprTS args)
     | _ -> 
         UnknownTS
 
 let rec uncheckedTypeTS g ty =
-    if isRefTupleTy g ty then 
-        let tys = destRefTupleTy g ty 
+    if isAnyTupleTy g ty then 
+        let tys =
+            if isRefTupleTy g ty then
```

Comment:
Those 4 lines are introduced 3 times. Better to put that into a destAnyTuppeTy

Reply:
yerp.

`destAnyTuplé`